dist/types/data-reference.d.ts

```diff
@@ -165,28 +165,28 @@ export declare class DataReference {
      * @param childPath Child key, index or path
      * @returns reference to the child
      */
-    child(childPath: string | number): DataReference;
+    child<Child = any>(childPath: string | number): DataReference<Child>;
     /**
      * Sets or overwrites the stored value
      * @param value value to store in database
      * @param onComplete optional completion callback to use instead of returning promise
      * @returns promise that resolves with this reference when completed
      */
-    set(value: any, onComplete?: (err: Error, ref: DataReference) => void): Promise<DataReference>;
+    set(value: T, onComplete?: (err: Error, ref: DataReference) => void): Promise<this>;
     /**
      * Updates properties of the referenced node
      * @param updates object containing the properties to update
      * @param onComplete optional completion callback to use instead of returning promise
      * @return returns promise that resolves with this reference once completed
      */
-    update(updates: object, onComplete?: (err: Error, ref: DataReference) => void): Promise<DataReference>;
+    update(updates: T, onComplete?: (err: Error, ref: DataReference) => void): Promise<this>;
```

Comment:
Forcing these methods to use `T` seemed to make sense because you wouldn't want it to be easy to ad-hoc change the value type of a `DataReference` that already has one set. It'll default to `any` if a type parameter hasn't been specified for the `DataReference` anyways.

Reply:
The type for update should be `Partial<T>` because you can pass in an object that only have the properties to update

Reply:
Whoops, will fix that!